File: internal/model/reply.go
package model

import "time"

// AutoReplyRule 自动回复规则
// 对应表: auto_reply_rules
type AutoReplyRule struct {
	ID            int64      `db:"id"`
	AccountID     int64      `db:"account_id"`
	Keyword       *string    `db:"keyword"`
	MatchType     int16      `db:"match_type"`
	ReplyType     int16      `db:"reply_type"`
	ReplyContent  string     `db:"reply_content"`
	ReplyTitle    *string    `db:"reply_title"`
	ReplyDesc     *string    `db:"reply_desc"`
	ReplyCoverURL *string    `db:"reply_cover_url"`
	ReplyURL      *string    `db:"reply_url"`
	Status        int16      `db:"status"`
	SortOrder     int        `db:"sort_order"`
	DeletedAt     *time.Time `db:"deleted_at"`
	CreatedAt     time.Time  `db:"created_at"`
	UpdatedAt     time.Time  `db:"updated_at"`
}

File: web/admin/src/api/reply.ts
import client from './client';

export interface ReplyVO {
  id: number;
  account_id: number;
  keyword?: string;
  match_type: number; // 0=精确 1=模糊
  reply_type: number; // 1=文本 2=图文
  reply_content: string;
  reply_title?: string;
  reply_desc?: string;
  reply_cover_url?: string;
  reply_url?: string;
  status: number; // 0=停用 1=启用
  sort_order: number;
}

export interface CreateReplyParams {
  keyword?: string;
  match_type: number;
  reply_type: number;
  reply_content: string;
  reply_title?: string;
  reply_desc?: string;
  reply_cover_url?: string;
  reply_url?: string;
  status?: number;
  sort_order?: number;
}

export interface UpdateReplyParams {
  keyword?: string;
  match_type?: number;
  reply_type?: number;
  reply_content?: string;
  reply_title?: string;
  reply_desc?: string;
  reply_cover_url?: string;
  reply_url?: string;
  status?: number;
  sort_order?: number;
}

// List reply rules by account
export function listReplies(accountId: number) {
  return client.get<{ code: number; msg: string; data: ReplyVO[] }>(`/accounts/${accountId}/replies`);
}

// Create a reply rule
export function createReply(accountId: number, data: CreateReplyParams) {
  return client.post<{ code: number; msg: string; data: ReplyVO }>(`/accounts/${accountId}/replies`, data);
}

// Update a reply rule
export function updateReply(ruleId: number, data: UpdateReplyParams) {
  return client.put<{ code: number; msg: string; data: ReplyVO }>(`/replies/${ruleId}`, data);
}

// Delete a reply rule
export function deleteReply(ruleId: number) {
  return client.delete<{ code: number; msg: string; data: null }>(`/replies/${ruleId}`);
}
